enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,690 @@
+* As a cluster administrator, I need to be able to configure static IPs for my tenant workers, since DHCP server is not available in my network.
+
+### Goals
+
+- Enable static IP address assignment for HyperShift guest cluster nodes running on KubeVirt
+- Provide a flexible API for defining IP address pools with support for various IP range formats (CIDR, hyphenated ranges, single IPs)
+- Support both IPv4 and IPv6 addressing for interface, default gw and DNS resolver.
+- Support KubeVirt VMs live migration so the assigned network configuration stick to the VMs after live migrating them.
+- Maintain the IPs across node or control plane pods restarts.
+
+### Non-Goals
+
+- Replacing existing network configuration mechanisms in standalone OpenShift clusters
+- Implementing a full-featured IPAM solution with conflict detection across multiple management clusters
+- Providing network configuration for components outside of HyperShift guest cluster nodes
+- Support complex network configurations like bonding of interfaces and others usually configured with ignition
+- Add IPAM network configuration to secondary networks not related to bootstrap process
+- Supporting dynamic IP pool expansion or runtime IP reallocation
+
+## Proposal
+
+This proposal introduces static IPAM capabilities through modifications to three key components:
+
+1. **Cluster API Provider KubeVirt**: Extend the API to support IP pool definitions and network configuration, implement IP allocation logic from defined pools
+2. **HyperShift**: Add network configuration options to the KubeVirt platform specification at nodepool, implement translation from this network configuration to capk ip pool and openstack network config
+3. **CoreOS Afterburn**: Enable parsing and application of OpenStack or netplan network config standard data as dracut network kernel args from cloud-init (config drive or nocloud) for KubeVirt VMs, this is similar what is done at [proxmoxve provider](https://github.com/coreos/afterburn/pull/1023).
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a HyperShift hosted cluster.
+
+1. Define the `NodePool` kubevirt platfrom configuration with the following:
+   - Disable the default pod network and select the network to replace it at the `NodePool` CRD instance:
+   ```yaml
+   spec:
+     plaform:
+       kubevirt:
+         attachDefaultNetwork: false
+         additionalNetworks:
+         - name: my-nad-1
+   ```
+
+   - The cluster administrator defines an IP pool configuration in the `NodePool` CRD instance:
+   ```yaml
+   spec:
+     plaform:
+       kubevirt:
+         bootstrapNetworkConfig:
+            network: my-nad-1-network  # network name (to assign a ip pool) not the nad name
+	        addresses: # IP address ranges in CIDR notation, hyphenated format, or as individual IPs
+            - 192.168.1.10-192.168.1.50
```

Comment:
Maybe I'll find an answer further down, but could you clarify what this list does? Will it assign one IP from each item? Will it combine all IPv4 items and all IPv6 and then assign one IPv4 and one IPv6 address? etc.

Reply:
> Maybe I'll find an answer further down, but could you clarify what this list does? Will it assign one IP from each item? Will it combine all IPv4 items and all IPv6 and then assign one IPv4 and one IPv6 address? etc.

The idea is to support same syntax as metallb, since customer is ok with that, this way we have a battle tested formated, do this make sense ?